src/components/add-products/add-products.js

```diff
@@ -1,38 +1,58 @@
 import React, { useState } from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
 
+import { productActions } from '../../store/products/action';
 import BuildForm from '../form-builder';
 import ProductCard from './product-card';
-import { productActions } from '../../store/products/action';
 
 import './product.scss';
 
 function AddProducts() {
+  const id = Number(useParams().id);
+  const products = useSelector(({ products }) => products.products);
+  let product = null;
```

Comment:
can these be a const instead of a let

Reply:
product can change from null to an object containing the product details.